pkg/publish/default.go

```diff
@@ -129,7 +133,7 @@ func pushResult(ctx context.Context, tag name.Tag, br build.Result, opt []remote
 		ociOpts := []ociremote.Option{ociremote.WithRemoteOptions(opt...)}
 
 		// Respect COSIGN_REPOSITORY
-		targetRepoOverride, err := ociremote.GetEnvTargetRepository()
```

Comment:
Does this mean we _already_ support `COSIGN_REPOSITORY`? 🤯 

If so, maybe we should just use this for now and document a plan to support non-`COSIGN`-named env vars in the future.

Reply:
>Does this mean we already support COSIGN_REPOSITORY? 🤯

Haha, yes 🫣

>If so, maybe we should just use this for now and document a plan to support non-COSIGN-named env vars in the future.

As you wish, but this keychain problem remains the same, at least we should fix this.

Reply:
Ah I get what you're saying now. Yeah, instead of taking an auth we should take a keychain, since we might be pushing to two repos/registries due to COSIGN_REPOSITORY.

Thanks for trying this out and for figuring out the fix. 👍

Reply:
>instead of taking an auth we should take a keychain

Exactly!